src/history.coffee

```diff
@@ -36,7 +37,9 @@ class History extends Serializable
       if patch instanceof BufferPatch
         @clearRedoStack()
     else
-      @undoStack.push(patch)
```

Comment:
We now wrap everything in a transaction so the history always contains the same kind of object. We thought it was required for our first attempt at fixing this, but it ended up being optional. Still, it's a structural improvement so we decided to leave it.